category_encoders/hashing.py

```diff
@@ -129,87 +132,7 @@ def __init__(self, max_process=0, max_sample=0, verbose=0, n_components=8, cols=
     def _fit(self, X, y=None, **kwargs):
         pass
 
-    def require_data(self, data_lock, new_start, done_index, hashing_parts, process_index):
-        is_finished = False
-        while not is_finished:
-            if data_lock.acquire():
-                if new_start.value:
-                    end_index = 0
-                    new_start.value = False
-                else:
-                    end_index = done_index.value
-
-                if all([self.data_lines > 0, end_index < self.data_lines]):
-                    start_index = end_index
-                    if (self.data_lines - end_index) <= self.max_sample:
-                        end_index = self.data_lines
-                    else:
-                        end_index += self.max_sample
-                    done_index.value = end_index
-                    data_lock.release()
-
-                    data_part = self.X.iloc[start_index: end_index]
-                    # Always get df and check it after merge all data parts
-                    data_part = self.hashing_trick(X_in=data_part, hashing_method=self.hash_method,
-                                                   N=self.n_components, cols=self.cols)
-                    part_index = int(math.ceil(end_index / self.max_sample))
-                    hashing_parts.put({part_index: data_part})
-                    is_finished = end_index >= self.data_lines
-                    if self.verbose == 5:
-                        print(f"Process - {process_index} done hashing data : {start_index} ~ {end_index}")
-                else:
-                    data_lock.release()
-                    is_finished = True
-            else:
-                data_lock.release()
-
-    def _transform(self, X):
-        """
-        Call _transform_single_cpu() if you want to use single CPU with all samples
-        """
-        self.X = X
-
-        self.data_lines = len(self.X)
-
-        data_lock = multiprocessing.Manager().Lock()
-        new_start = multiprocessing.Manager().Value('d', True)
-        done_index = multiprocessing.Manager().Value('d', int(0))
-        hashing_parts = multiprocessing.Manager().Queue()
-
-        if self.auto_sample:
```

Comment:
you don't need the auto sample attribute anymore

Reply:
Removing it!